src/dask_histogram/core.py

```diff
@@ -193,18 +193,37 @@ def _blocked_df_w_s(
     )
 
 
-def _blocked_dak(data: Any, *, histref: bh.Histogram | None = None) -> bh.Histogram:
+def _blocked_dak(
+    data: Any,
+    weights: Any | None,
+    sample: Any | None,
```

Comment:
This is a bit more elegant than having a single dedicated function for each arity!